lib/src/default_index_store.rs

```diff
@@ -1825,6 +1841,7 @@ mod tests {
         } else {
             Box::new(index)
         };
+        let index = as_composite(index.as_ref());
```

Comment:
`index` could be cast to `Box<dyn IndexSegment>` upfront, but I have no idea which is better.

Reply:
Good point. That's simpler, so I changed it. Thanks for the suggestion.